### PR TITLE
Fix hbonds path

### LINKED ISSUE #4285: hydrogenbonds docs still reference hbonds path
This mailing issue shows that our `stable` docs still reference the `hbonds` path:

https://groups.google.com/d/msgid/mdnalysis-discussion/CANM7oPh0ogkSZBY7Gq%3D6nyeBkKXCnTCQEbwvjumO3j7AvS72jQ%40mail.gmail.com?utm_medium=email&utm_source=footer

### PR DESCRIPTION
**Pull request by lilyminium**
Fixes #4285

Changes made in this Pull Request:
 - 

PR Checklist
------------
 - [ ] Tests?
 - [x] Docs?
 - [ ] CHANGELOG updated?
 - [ ] Issue raised/referenced?

## Developers certificate of origin
- [x] I certify that this contribution is covered by the LGPLv2.1+ license as defined in our [LICENSE](https://github.com/MDAnalysis/mdanalysis/blob/develop/LICENSE) and adheres to the [**Developer Certificate of Origin**](https://developercertificate.org/).

----
:books: Documentation preview :books:: https://mdanalysis--4286.org.readthedocs.build/en/4286/

### CODE REVIEW

File: package/CHANGELOG [hunk 1]
```diff
@@ -13,7 +13,8 @@ The rules for this file:
   * release numbers follow "Semantic Versioning" http://semver.org
 
 ------------------------------------------------------------------------------
-??/??/?? IAlibay, ianmkenney, PicoCentauri, pgbarletta, p-j-smith, richardjgowers
+??/??/?? IAlibay, ianmkenney, PicoCentauri, pgbarletta, p-j-smith, richardjgowers,
+          lilyminium
```
Comment (RMeli) [inline package/CHANGELOG]:
```suggestion
         lilyminium
```

**Review by RMeli — APPROVED** [review]

**Review by RMeli — CHANGES_REQUESTED** [review]
I think the duecredit test needs to be updated as well.

**Review by RMeli — APPROVED** [review]
LGTM, thanks @lilyminium

**Comment by lilyminium** [comment]
Thanks for the review @RMeli!

**Pull request merged**